fix: Added Pitstops, Qualifying, Circuits, Constructors, Drivers Standings and Constructors standings endpoint docs

MattMaginniss opened the pull request:
## Why are you making this change?

To add documentation for the `/pitstops`, `/qualifying`, `/circuits`, `/constructors`, `/driverStandings`, and `/constructorStandings` endpoints  

## Contributing Checklist
- [ ] Unit tests for the changes are included in this PR.
- [x] I have read and agreed to the [contributing guidelines](CONTRIBUTING.md).

jolpica commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,178 @@
+
+**Note**: To utilize the `season` parameter it needs to be the first route after `/ergast/f1/`
+
+---
```
Not just this endpoint, but the Round parameter is also available for all endpoints.

e.g. `/ergast/f1/2024/5/constructors/`, `/ergast/f1/2024/5/driverstandings/`, etc

jolpica commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,178 @@
+
+Gets list of constructors 
+
+**URL** : `/ergast/f1/constructors`
```
Small nitpick, all urls need to end with a `/` or `.json` as the framework automatically redirects urls without a `/` or `.` to one with one.

So `/ergast/f1/constructors/` or `/ergast/f1/constructors.json`

jolpica commented on docs/endpoints/constructorStandings.md:
```diff
@@ -0,0 +1,205 @@
+
+### Round
+
+**Filters for the constructors standings after a specified round in a specific season. Round numbers 1 -> n races are valid as well as 'last'**
```
```suggestion
**Filters for the constructors standings after a specified round in a specific season. Round numbers 1 -> n races are valid as well as `next` and `last`**
```

MattMaginniss replied:
I fixed the formatting for `last`

but How do we feel about how "next" for example, When we run next on the current season it returns an empty StandingsTable.StandingsList like this:

```
{
  "MRData": {
    "xmlns": "",
    "series": "f1",
    "url": "http://api.jolpi.ca/ergast/f1/2024/next/constructorstandings/",
    "limit": "30",
    "offset": "0",
    "total": "0",
    "StandingsTable": {
      "season": "2024",
      "round": "18",
      "StandingsLists": []
    }
  }
}
```

and if you use `next` on a specific past season like 2018 it will just return the standings after the first round of the 2019 season. While technically it is valid as there is not a server error  but I don't know if it is something that makes sense to list a feature?

Happy to add it. Just wanted to present that thought.

jolpica replied:
Yeah you're totally right, `next` will never make sense on the driver/constructor standing endpoints.
So I think you're right about not mentioning it.

jolpica commented on docs/endpoints/constructorStandings.md:
```diff
@@ -0,0 +1,205 @@
+
+**Filters for the constructors standings after a specified round in a specific season. Round numbers 1 -> n races are valid as well as `last`**
+
+`/{season}/{roundNumber}/` -> ex: `/ergast/f1/2024/5/constructorstandings/`
```
I'm open to suggestions on what it should be, but we should name things consistently, e.g. `round` vs `roundNumber`, `season` vs `YEAR`.

For inspiration, this is what the value are called internally. We don't necessarily have to use them
https://github.com/jolpica/jolpica-f1/blob/8424ed415102dfa24de5b8a85faf605e9a5ee995/jolpica_api/ergastapi/urls.py#L44-L55

Not sure, for example, whether in driver filters we should use `driver`, `driver_ref` or `driver_reference`,

MattMaginniss replied:
Honestly last night I was slightly struggling with a few things like "what did I call it there" happy to refer to it as something more consistent.

jolpica replied:
Do what you think is best, I'll find some time to do a full read through of everything this evening.

MattMaginniss replied:
Hey @jolpica I found my little notepad I was scribbling thoughts onto. 

The reason I was using driverId for the field after drivers/ is because if you use a drivers filter. What is shown in the response for: https://api.jolpi.ca/ergast/f1/drivers/hamilton/races/ is this:
<img width="229" alt="image" src="https://github.com/user-attachments/assets/285bc8ae-f62a-46e1-9ae6-694bf1f35d66">

So I was using driverId, constructorId, season, etc because this is the name of the property that is shown in the response.

As for fastest, grid, results and status I slightly modified them because the fields are technically just called "fastest", "grid", "position" and "status" respectively and that doesn't properly capture what the field is meant oto mean.

So I changed:
|route|from|to|
|----|-----|-----|
|/fastest   |  `/{fastest}`    |    `/{lapRank}` |
|/grid     |     `/{grid}`         |   `/{gridPosition}` |
|/results    |  `/{position}`  |   `/{finishPosition}`|
|/status    |   `/{status}`     |   `/{statusId}`|

as I felt they are a bit more clear as to what they do/are at a simple glance. (like if you saw `status` you might think "Disqualified", or "Accident" would be valid, but `statusId` should tell you that you need to use a specific ID for a status.

If someone were to look for the status in the `/status/` endpoint they will see the status objects where there is a field called `statusId`and `status` and the `status`field here is actually not valid. 

You cant do: https://api.jolpi.ca/ergast/f1/status/Disqualified/races

but you can: https://api.jolpi.ca/ergast/f1/status/2/races

```
"StatusTable": {
      "Status": [
        {
          "statusId": "1",
          "count": "7592",
          "status": "Finished"
        },
        {
          "statusId": "2",
          "count": "146",
          "status": "Disqualified"
        },
        {
          "statusId": "3",
          "count": "1059",
          "status": "Accident"
        },
```

jolpica commented on docs/README.md:
I'm seeing a bit of duplication of some filters such as round, season & constructors.
What do you think about common pages? Or do you think theres enough differences to keep them separate?

MattMaginniss replied:
There is almost identical duplication on season and round but they are optional for some and required for others.

I don't know if we have a single filter that is valid, makes sense and is either optional or required for every endpoint and that's basically my main concern. 

We could link to a route parameter doc that holds all of them and for each specific endpoint list which ones X,Y,Z are available(with links to the common page for each) and then mark if Y is required for that specific one. This could be nice but I want to make sure it still provides good understanding of what it means for each endpoint.

My concern is if you are not familiar with the common filters you need to jump between two pages  to see and understand what's available and what it does for a specific endpoint. 

I will do some looking later to see what it looks like pulling them out to a common file and see what the actual overlap is and if it makes sense.

MattMaginniss commented:
Added Pitstops, Circuits, Constructors, Drivers Standings and Constructors Standings. and made the changes requested in the previous reviews (including adding the trailing / to the endpoints in the docs.)

I am done for now. so just requested a re-review. 

I have changed to use season more consistently instead of year as well as round vs roundNumber as but left some as my reasoning for here: https://github.com/jolpica/jolpica-f1/pull/83#discussion_r1769656600

jolpica commented on docs/endpoints/circuits.md:
```diff
@@ -0,0 +1,209 @@
+
+**Response Fields** :
+
+[Common Response Fields](./README.md#common-response-fields)
```
```suggestion
[Common Response Fields](docs/README.md#common-response-fields)
```
Link also doesn't work

jolpica commented on docs/endpoints/circuits.md:
```diff
@@ -0,0 +1,209 @@
+
+## Circuits Object Fields:
+
+|Field|Required|
```
```suggestion
|Field|Always Included|
```
Little nitpick, required just feels more like an ask of the user of the API

MattMaginniss replied:
Great idea. In an earlier comment (or maybe commit) I said I have no idea a better word for required but knew it wasn't the best.

jolpica commented on docs/endpoints/circuits.md:
```diff
@@ -0,0 +1,209 @@
+
+|Field|Required|
+|---|:---:|
+|circuitId| ✅ |
+|url|✅|
+|circuitName|✅|
+|Location|✅|
```
```suggestion
|circuitId| 🟡 |
|url|🟡|
|circuitName|✅|
|Location|🟡|
```
New option of yellow, may be best. We 99% of time provide this, but from a database perspective there could be some times where the fields are missing.

MattMaginniss replied:
Good point. I like the yellow instead

jolpica commented on docs/endpoints/circuits.md:
````diff
@@ -0,0 +1,209 @@
+```
+
+### Get all circuits which have had a race with 29 driver results.
+
+* Note this is missing Logan Sargent as he did not start the race even though he participated in the weekend.
````
```suggestion
```

jolpica commented on docs/endpoints/circuits.md:
```diff
@@ -0,0 +1,209 @@
+
+## Circuits Object Fields:
+
+|Field|Required|
```
```suggestion
|Field|Required|Description|type|
```
This is a note to self for a future PR / issue. I don't want to ask any more of you!

MattMaginniss replied:
:) Added

jolpica commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,188 @@
+
+**Filters for only a specified constructor**
+
+`/constructors/{constructorId}/` -> ex: `/ergast/f1/constructors/williams/constructors/`
```
```suggestion
`/constructors/{constructorId}/` -> ex: `/ergast/f1/constructors/williams/`
```

jolpica commented on docs/endpoints/constructorStandings.md:
```diff
@@ -0,0 +1,205 @@
+
+|Field|Required|
+|---|:---:|
+|position| ✅ |
+|positonText|✅|
+|points|✅|
+|wins|✅|
+|Constructor|✅|
```
```suggestion
|position| ❌ |
|positonText|✅|
|points|✅|
|wins|✅|
|Constructor|✅|
Possible values for positionText include: `E` Excluded (2007 McLaren), `D` Disqualified, `-` for ineligible or the position as a string otherwise.  
```

jolpica commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,188 @@
+
+|Field|Required|
+|---|:---:|
+|constructorId| ✅ |
+|url|✅|
+|name|✅|
+|nationality|✅|
```
```suggestion
|constructorId| 🟡 |
|url|🟡|
|name|✅|
|nationality|🟡|
```

jolpica reviewed (commented):
Thanks again for all your contributions on this, few things I've spotted.

Happy to merge this and address them myself later, or for you to fix.

chrisfree commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,188 @@
+
+### Season
+
+**Filters only constructors that participated in a specified season. Year numbers are valid as is `current` to get the current season list of constructors**
```
```suggestion
Filters only constructors that participated in a specified season. Year numbers are valid as is `current` to get the current season list of constructors.
```

chrisfree commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,188 @@
+
+`/{season}/` -> ex: `/ergast/f1/2024/constructors/`
+
+**Note**: To utilize the `season` parameter it needs to be the first route after `/ergast/f1/`
```
```suggestion
**Note**: To utilize the `season` parameter it needs to be the first argument after `/ergast/f1/`.
```

chrisfree commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,188 @@
+
+### Round
+
+**Filters only constructors that participated in a specified round of a specific season. Round numbers 1 -> n races are valid as well as `last`**
```
```suggestion
Filters only constructors that participated in a specified round of a specific season. Round numbers 1 -> `n` races are valid as well as `last`.
```

chrisfree commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,188 @@
+
+`/{round}/` -> ex: `/ergast/f1/2024/1/constructors/`
+
+**Note**: To utilize the `round` parameter it needs to be used with the `season` filter and be the first route after `/ergast/f1/{season}/`
```
```suggestion
**Note**: To utilize the `round` parameter it needs to be used with the `season` filter and be the first argument after `/ergast/f1/{season}/`
```

chrisfree commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,188 @@
+
+### circuits
+
+**Filters for only constructors who have participated in a race at a given circuit**
```
```suggestion
Filters for only constructors who have participated in a race at a given circuit.
```

chrisfree commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,188 @@
+
+### constructors
+
+**Filters for only a specified constructor**
```
```suggestion
Filters for only a specified constructor.
```

chrisfree commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,188 @@
+
+### drivers
+
+**Filters for only constructors that had a driver race for them**
```
```suggestion
Filters for only constructors that had a driver race for them.
```

chrisfree commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,188 @@
+
+### fastest
+
+**Filters for only constructors that finished a race with a lap that was the ranked in the specified position**
```
```suggestion
Filters for only constructors that finished a race with a lap that was the ranked in the specified position.
```

chrisfree commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,188 @@
+
+### grid
+
+**Filters for only constructors which had a driver racing for them start a race in a specific grid position**
```
```suggestion
Filters for only constructors which had a driver racing for them start a race in a specific grid position.
```

chrisfree commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,188 @@
+
+### results
+
+**Filters for only constructors which had a driver racing for them finish a race in a specific position**
```
```suggestion
Filters for only constructors which had a driver racing for them finish a race in a specific position.
```

chrisfree commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,188 @@
+
+### status
+
+**Filters for only constructors who had a driver finish a race with a specific statusId**
```
```suggestion
Filters for only constructors who had a driver finish a race with a specific `statusId`.
```

chrisfree commented on docs/endpoints/constructors.md:
```diff
@@ -0,0 +1,188 @@
+
+`MRData.ConstructorTable.Constructors` : The list of all constructors returned.
+
+`MRData.ConstructorTable.Constructors[i]` : A given constructor object
```
```suggestion
`MRData.ConstructorTable.Constructors[i]` : A given constructor object.
```

chrisfree commented on docs/endpoints/driverStandings.md:
```diff
@@ -0,0 +1,216 @@
+[← Documentation Home](/docs/README.md)
+# Driver Standings
+
+Gets a season's drivers standings from first to last place. 
```
```suggestion
Returns a season's drivers standings from first to last place. 
```

chrisfree commented on docs/endpoints/driverStandings.md:
```diff
@@ -0,0 +1,216 @@
+
+### Season \*\*REQUIRED\*\*
+
+**Filters for the drivers standing of a specified season. Year numbers are valid as is `current` to get the current seasons drivers standings**
```
```suggestion
Filters for the drivers standing of a specified season. Year numbers are valid as is `current` to get the current seasons drivers standings.
```

chrisfree commented on docs/endpoints/driverStandings.md:
```diff
@@ -0,0 +1,216 @@
+
+### Round
+
+**Filters for the drivers standings after a specified round in a specific season. Round numbers 1 -> n races are valid as well as `last`**
```
```suggestion
Filters for the drivers standings after a specified round in a specific season. Round numbers 1 -> `n` races are valid as well as `last`.
```

chrisfree commented on docs/endpoints/driverStandings.md:
```diff
@@ -0,0 +1,216 @@
+
+`/{season}/{round}/` -> ex: `/ergast/f1/2024/5/driverstandings/`
+
+**Note**: To utilize the `round` parameter it must be combined with a season filter and needs to be the first route after `/ergast/f1/{season}/`
```
```suggestion
**Note**: To utilize the `round` parameter it must be combined with a season filter and needs to be the first argument after `/ergast/f1/{season}/`.
```

chrisfree commented on docs/endpoints/driverStandings.md:
```diff
@@ -0,0 +1,216 @@
+
+### drivers
+
+**Filters for only a specific driver's drivers standing information for a given year**
```
```suggestion
Filters for only a specific driver's drivers standing information for a given year.
```

chrisfree commented on docs/endpoints/driverStandings.md:
```diff
@@ -0,0 +1,216 @@
+
+### position
+
+**Filters for only the driver in a given position for a given year**
```
```suggestion
Filters for only the driver in a given position for a given year.
```

chrisfree commented on docs/endpoints/drivers.md:
```diff
@@ -1,47 +1,57 @@
 ### Season
 
-**Filters only drivers that participated in a specified season. Year numbers are valid as is 'current' to get the current season list of drivers**
+**Filters only drivers that participated in a specified season. Year numbers are valid as is `current` to get the current season list of drivers**
```
```suggestion
Filters only drivers that participated in a specified season. Year numbers are valid as is `current` to get the current season list of drivers.
```

chrisfree commented on docs/endpoints/drivers.md:
```diff
@@ -1,47 +1,57 @@
 # Drivers
 
-Gets list of drivers 
+Gets list of drivers in alphabetical order by driverId
```
```suggestion
Returns a list of drivers in alphabetical order by `driverId`.
```

chrisfree commented on docs/endpoints/drivers.md:
```diff
@@ -1,47 +1,57 @@
 
+### Round
+
+**Filters only drivers that participated in a specified round of a specific season. Round numbers 1 -> n races are valid as well as `last`**
```
```suggestion
Filters only drivers that participated in a specified round of a specific season. Round numbers 1 -> `n` races are valid as well as `last`.
```

chrisfree commented on docs/endpoints/drivers.md:
```diff
@@ -1,47 +1,57 @@
+
+`/{round}/` -> ex: `/ergast/f1/2024/1/drivers/`
+
+**Note**: To utilize the `round` parameter it needs to be used with the `season` filter and be the first route after `/ergast/f1/{season}`
```
```suggestion
**Note**: To utilize the `round` parameter it needs to be used with the `season` filter and be the first argument after `/ergast/f1/{season}`.
```

chrisfree commented on docs/endpoints/drivers.md:
```diff
@@ -1,47 +1,57 @@
+
 ### circuits
 
 **Filters for only drivers who have participated in a race at a given circuit**
```
```suggestion
Filters for only drivers who have participated in a race at a given circuit.
```

chrisfree commented on docs/endpoints/drivers.md:
```diff
@@ -1,47 +1,57 @@
 
 ### drivers
 
 **Filters for only drivers that match the specific driverId**
```
```suggestion
Filters for only drivers that match the specific `driverId`.
```

chrisfree commented on docs/endpoints/drivers.md:
```diff
@@ -50,7 +60,7 @@ Gets list of drivers
 
 **Filters for only drivers that finished a race with a lap that was the ranked in the specified position**
```
```suggestion
Filters for only drivers that finished a race with a lap that was the ranked in the specified position.
```

chrisfree commented on docs/endpoints/drivers.md:
```diff
@@ -59,23 +69,23 @@ Gets list of drivers
 
 **Filters for only drivers who has started a race in a specific grid position**
```
```suggestion
Filters for only drivers who have started a race in a specific grid position.
```

chrisfree commented on docs/endpoints/drivers.md:
```diff
@@ -59,23 +69,23 @@ Gets list of drivers
 
 ### results
 
 **Filters for only drivers who has finished a race in a specific position**
```
```suggestion
Filters for only drivers who have finished a race in a specific position.
```

chrisfree commented on docs/endpoints/drivers.md:
```diff
@@ -59,23 +69,23 @@ Gets list of drivers
 
 ### status
 
 **Filters for only drivers who have finished a race with a specific statusId**
```
```suggestion
Filters for only drivers who have finished a race with a specific `statusId`.
```

chrisfree commented on docs/endpoints/pitstops.md:
```diff
@@ -0,0 +1,242 @@
+[← Documentation Home](/docs/README.md)
+# Pitstops Standings
+
+Gets a given races list of pitstops in from earliest to latest `time` in which the pitstop occurred.
```
```suggestion
Returns a given races list of pitstops,  from earliest to latest `time` in which the pitstop occurred.
```

chrisfree commented on docs/endpoints/pitstops.md:
```diff
@@ -0,0 +1,242 @@
+
+## Route Parameters:
+
+### Season \*\*REQUIRED\*\*
```
```suggestion
### Season (required)
```

chrisfree commented on docs/endpoints/pitstops.md:
```diff
@@ -0,0 +1,242 @@
+
+### Season \*\*REQUIRED\*\*
+
+**Filters for the season that the list of pitstops will be from. Year numbers are valid as is `current` to get the pitstops of a given round in the current season**
```
```suggestion
Filters for the season that the list of pitstops will be from. Year numbers are valid as is `current` to get the pitstops of a given round in the current season.
```

chrisfree commented on docs/endpoints/pitstops.md:
```diff
@@ -0,0 +1,242 @@
+
+`/{season}/` -> ex: `/ergast/f1/2024/1/pitstops/`
+
+**Note**: To utilize the `season` parameter it needs to be the first route after `/ergast/f1/`
```
```suggestion
**Note**: To utilize the `season` parameter it needs to be the first argument after `/ergast/f1/`.
```

chrisfree commented on docs/endpoints/pitstops.md:
```diff
@@ -0,0 +1,242 @@
+
+### Round \*\*REQUIRED\*\*
+
+**Filters for the round in a specific season that the list of pitstops will be from. Round numbers 1 -> n races are valid as well as `last`**
```
```suggestion
Filters for the round in a specific season that the list of pitstops will be from. Round numbers 1 -> `n` races are valid as well as `last`.
```

chrisfree commented on docs/endpoints/pitstops.md:
```diff
@@ -0,0 +1,242 @@
+
+`/{season}/{round}/` -> ex: `/ergast/f1/2024/5/pitstops/`
+
+**Note**: To utilize the `round` parameter it must be combined with a season filter and needs to be the first route after `/ergast/f1/{season}/`
```
```suggestion
**Note**: To utilize the `round` parameter it must be combined with a season filter and needs to be the first argument after `/ergast/f1/{season}/`.
```

chrisfree commented on docs/endpoints/pitstops.md:
```diff
@@ -0,0 +1,242 @@
+
+### drivers
+
+**Filters for only for a specific drivers's list of pitstops in a seasons round**
```
```suggestion
Filters for only for a specific drivers's list of pitstops in a season's round.
```

chrisfree commented on docs/endpoints/pitstops.md:
```diff
@@ -0,0 +1,242 @@
+
+### laps
+
+**Filters for only pitstops that took place in a given lap of a race**
```
```suggestion
Filters for only pitstops that took place in a given lap of a race.
```

chrisfree commented on docs/endpoints/races.md:
```diff
@@ -1,47 +1,57 @@
 # Races
 
-Gets list of races 
+Gets list of races from earliest to latest.
```
```suggestion
Returns a list of races from earliest to latest.
```

chrisfree commented on docs/endpoints/races.md:
```diff
@@ -1,47 +1,57 @@
 ### Season
 
-**Filters for races only from a specified season. Year numbers are valid as is 'current' to get the current season**
+**Filters for races only from a specified season. Year numbers are valid as is `current` to get the current season**
```
```suggestion
Filters for races only from a specified season. Year numbers are valid as is `current` to get the current season.
```

chrisfree commented on docs/endpoints/races.md:
```diff
@@ -1,47 +1,57 @@
-`/{season}/` -> ex: `/ergast/f1/2024/races`
+`/{season}/` -> ex: `/ergast/f1/2024/races/`
 
 **Note**: To utilize the `season` parameter it needs to be the first route after `/ergast/f1/`
```
```suggestion
**Note**: To utilize the `season` parameter it needs to be the first argument after `/ergast/f1/`.
```

chrisfree commented on docs/endpoints/races.md:
```diff
@@ -1,47 +1,57 @@
 
+### Round
+
+**Filters for the race for a specified round in a specific season. Round numbers 1 -> n races are valid as well as `last` and `next`**
```
```suggestion
Filters for the race for a specified round in a specific season. Round numbers 1 -> `n` races are valid as well as `last` and `next`.
```

chrisfree commented on docs/endpoints/races.md:
```diff
@@ -1,47 +1,57 @@
+
+`/{season}/{round}/` -> ex: `/ergast/f1/2024/5/races/`
+
+**Note**: To utilize the `round` parameter it must be combined with a season filter and needs to be the first route after `/ergast/f1/{season}/`
```
```suggestion
**Note**: To utilize the `round` parameter it must be combined with a season filter and needs to be the first argument after `/ergast/f1/{season}/`.
```

chrisfree commented on docs/endpoints/races.md:
```diff
@@ -1,47 +1,57 @@
+
 ### circuits
 
 **Filters for only races featuring a specified circuit**
```
```suggestion
Filters for only races featuring a specified circuit.
```

chrisfree commented on docs/endpoints/races.md:
```diff
@@ -1,47 +1,57 @@
 
 ### constructors
 
 **Filters for only races featuring a specified constructor**
```
```suggestion
Filters for only races featuring a specified constructor.
```

chrisfree commented on docs/endpoints/races.md:
```diff
@@ -1,47 +1,57 @@
 
 ### drivers
 
 **Filters for only races featuring a specified driver**
```
```suggestion
Filters for only races featuring a specified driver.
```

chrisfree commented on docs/endpoints/races.md:
```diff
@@ -50,15 +60,15 @@ Gets list of races
 
 ### status
 
 **Filters for only races featuring a specified finishing status of a driver**
```
```suggestion
Filters for only races featuring a specified finishing status of a driver.
```

chrisfree commented on docs/endpoints/races.md:
```diff
@@ -80,29 +90,29 @@ Gets list of races
 ### Get list of races in F1 history
 
-`https://api.jolpi.ca/ergast/f1/races/`
+`https://api.jolpi.ca/ergast/f1/races//`
```
```suggestion
`https://api.jolpi.ca/ergast/f1/races/`
```

chrisfree commented on docs/endpoints/seasons.md:
```diff
@@ -1,21 +1,21 @@
 ### Season
 
-**Filters for a specified season. Year numbers are valid as is 'current' to get the current season**
+**Filters for a specified season. Year numbers are valid as is `current` to get the current season**
```
```suggestion
Filters for a specified season. Year numbers are valid as is `current` to get the current season.
```

chrisfree commented on docs/endpoints/seasons.md:
```diff
@@ -1,21 +1,21 @@
-`/{season}/` -> ex: `/ergast/f1/2024/seasons`
+`/{season}/` -> ex: `/ergast/f1/2024/seasons/`
 
 **Note**: To utilize the `season` parameter it needs to be the first route after `/ergast/f1/`
```
```suggestion
**Note**: To utilize the `season` parameter it needs to be the first route after `/ergast/f1/`.
```

chrisfree commented on docs/endpoints/seasons.md:
```diff
@@ -25,23 +25,23 @@ Gets list of seasons
 
 **Filters for only seasons featuring a specified circuit**
```
```suggestion
Filters for only seasons featuring a specified circuit.
```

chrisfree commented on docs/endpoints/seasons.md:
```diff
@@ -25,23 +25,23 @@ Gets list of seasons
 
 ### constructors
 
 **Filters for only seasons featuring a specified constructor**
```
```suggestion
Filters for only seasons featuring a specified constructor.
```

chrisfree commented on docs/endpoints/seasons.md:
```diff
@@ -25,23 +25,23 @@ Gets list of seasons
 
 ### drivers
 
 **Filters for only seasons featuring a specified driver**
```
```suggestion
Filters for only seasons featuring a specified driver.
```

chrisfree commented on docs/endpoints/seasons.md:
```diff
@@ -50,15 +50,15 @@ Gets list of seasons
 
 **Filters for only seasons featuring a specified grid position**
```
```suggestion
Filters for only seasons featuring a specified grid position.
```

chrisfree commented on docs/endpoints/seasons.md:
```diff
@@ -50,15 +50,15 @@ Gets list of seasons
 
 ### status
 
 **Filters for only seasons featuring a specified finishing status of a driver in at least one race that season**
```
```suggestion
Filters for only seasons featuring a specified finishing status of a driver in at least one race that season.
```

chrisfree reviewed (dismissed):
@MattMaginniss this is so appreciated. I gave this a simple, high-level review. I focused a bit on uniformity with periods and dropped some formatting that I thought was unnecessary.

Thank you very much for helping kickstart the docs! 👏🏻

MattMaginniss commented on docs/endpoints/circuits.md:
```diff
@@ -0,0 +1,207 @@
+
+### fastest
+
+**Filters for only circuits that have had a race with a lap that was the ranked in the specified position**
```
To be honest @chrisfree, it's such an obtuse filter.

What it does, is it returns a list of circuits where there has been a race where a lap has been classified as the nth fastest lap of that given race.

I have changed it to this:

> Filters for a list of circuits where a race finished with a driver completing a lap that was the ranked in the specified position.

MattMaginniss commented:
> @MattMaginniss this is so appreciated. I gave this a simple, high-level review. I focused a bit on uniformity with periods and dropped some formatting that I thought was unnecessary.
> 
> Thank you very much for helping kickstart the docs! 👏🏻

Yeah, @chrisfree I think the removal of the extra bolded headlines and what not all make sense to remove. I have updated all the things. Thanks for the suggestion on the english/grammar bits.

Happy to help. One big complaint I had with ergast was that the docs were a bit lacking so would love to make this a bit more friendly for new people to joining or wanting to use the API.

jolpica commented on docs/endpoints/races.md:
```diff
@@ -74,27 +84,27 @@ Gets list of races
+|ThirdPractice|🟡|Third Practice (date, time)|Object
+|Qualifying|🟡|Qualifying (date, time)|Object
+|Sprint|🟡|Sprint Race (date, time)|Object
+|SprintQualifying|🟡|Sprint Qualifying (date, time)|Object
```
Also can have a `SprintShootout` field (I believe 2023).

jolpica replied:
```suggestion
|SprintQualifying / SprintShootout|🟡| Shootouts took place in 2023, otherwise they are Qualifying (date, time)|Object
```

jolpica commented on docs/endpoints/qualifying.md:
```diff
@@ -0,0 +1,301 @@
+|Field|Always Included|Description|type
+|---|:---:|---|---|
+|number|✅|Driver's car number|String
+|position|✅|Qualifying Result Position|String
```
```suggestion
|position| 🟡 |Qualifying Result Position|String
```

jolpica commented on docs/endpoints/drivers.md:
```diff
@@ -91,30 +101,30 @@ Gets list of drivers
+|driverId|✅|Unique ID of the Driver|String
+|permanentNumber|🟡|Permanent Number assigned to the driver|String
+|code|🟡|Driver Code, usually 3 characters|String
+|url|✅|Wikipedia URL to the Drivers profile|String
+|givenName|✅|First name|String
+|familyName|✅|Last name|String
+|dateOfBirth|✅|Date of Birth (YYYY-MM-DD format)|String
+|nationality|✅|Nationality of Driver|String
```
```suggestion
|url| 🟡 |Wikipedia URL to the Drivers profile|String
|givenName|✅|First name|String
|familyName|✅|Last name|String
|dateOfBirth| 🟡 |Date of Birth (YYYY-MM-DD format)|String
|nationality| 🟡 |Nationality of Driver|String
```

jolpica reviewed (approved):
This is a great improvement to the documentation of the API.
Thanks for the contribution!

MattMaginniss commented:
@jolpica Hey thanks for those last adjustments.
